content-security-policy/embedded-enforcement/subsumption_algorithm-general.html

```diff
@@ -0,0 +1,54 @@
+<!DOCTYPE html>
+<html>
+<head>
+<title>Embedded Enforcement: Subsumption Algorithm - Basic implementation.</title>
+  <script src="/resources/testharness.js"></script>
+  <script src="/resources/testharnessreport.js"></script>
+  <script src="../support/testharness-helper.sub.js"></script>
+</head>
+<body>
+  <script>
+    var tests = [
+      { "name": "If there is no required csp, iframe should load.", 
+        "required_csp": null, 
+        "returned_csp": null,
+        "expected": IframeLoad.EXPECT_LOAD },
+      { "name": "Iframe with empty returned CSP should be blocked.", 
+        "required_csp": "style-src 'none';", 
+        "returned_csp": null,
+        "expected": IframeLoad.EXPECT_BLOCK },
+      { "name": "Iframe with matching CSP should load.", 
+        "required_csp": "style-src 'none' 8 script-src 'unsafe-inline'", 
+        "returned_csp": "style-src 'none' 8 script-src 'unsafe-inline'", 
+        "expected": IframeLoad.EXPECT_LOAD },
+      { "name": "Iframe with more restricting CSP should load.", 
+        "required_csp": "script-src 'nonce-abc' 'nonce-123'", 
+        "returned_csp": "script-src 'nonce-abc'", 
+        "expected": IframeLoad.EXPECT_LOAD },
+      { "name": "Iframe with less restricting CSP should be blocked.", 
+        "required_csp": "style-src 'none' 8 script-src 'none'", 
+        "returned_csp": "style-src 'none' 8 script-src 'self'", 
```

Comment:
That seems very strange. I'll look at this with you when I get back to my desk. We shouldn't need to do this...

Are you running `encodeURIComponent` on the string you're passing into the URL?